submissions/api.py

```diff
@@ -196,6 +196,20 @@ def create_submission(student_item_dict, answer, submitted_at=None, attempt_numb
         raise SubmissionInternalError(error_message)
 
 
+def _get_submission_model(uuid, read_replica=False):
+    """
+    Helper to retrieve a given Submission object from the database. Helper is needed to centralize logic that fixes
+    EDUCATOR-1090, because uuids are stored both with and without hyphens.
+    """
+    submission_qs = Submission.objects
+    if read_replica:
+        submission_qs = _use_read_replica(submission_qs)
+
+    query_regex = "^{}$|^{}$".format(uuid, uuid.replace("-",""))
+    submission = submission_qs.get(uuid__regex=query_regex)
```

Comment:
Doing the regex forces string comparison on the database values, where uuid = A OR uuid = B will convert both A and B to the same (non-matching) formatted UUID hex value